lib/mastodon/cli/domains.rb

```diff
@@ -4,7 +4,90 @@
 require_relative 'base'
 
 module Mastodon::CLI
+  # This class is a helper so we can reuse the validation logic in
+  # Admin::Import, which typically expects ActionDispatch::Parameters as an
+  # argument
+  class DomainBlocklistData
+    attr_accessor :original_filename, :path
+
+    def self.from(filename, filepath)
+      import_data = new
+      import_data.original_filename = filename
+      import_data.path = filepath
+      import_data
+    end
+  end
+
+  class DomainBlocklist < Base
+    desc 'import [NAME] [URL]', 'Imports a blocklist from the given URL'
+    def import(name, url)
+      filename = "#{name}.csv"
+      tmpfile = Tempfile.new(filename)
+
+      begin
+        say('Fetching blocklist...')
+        response = HTTP.headers(
+          'Accept' => 'text/csv',
+          'User-Agent' => Mastodon::Version.user_agent
+        ).get(url)
+
+        unless response.code == 200
+          say("Error: Failed to fetch blocklist from: #{url}", :red)
+          exit(1)
+        end
+
+        tmpfile.write(response.body.to_s)
+
+        say('Fetched blocklist!')
+
+        import = Admin::Import.new(data: DomainBlocklistData.from(filename, tmpfile.path))
+
+        unless import.validate
+          import_errors = import.errors.full_messages.map { |err| "- #{err}" }.join("\n")
+
+          say("Error: Failed to import blocklist, details:\n#{import_errors}", :red)
+          exit(1)
+        end
+
+        say("Importing #{import.csv_rows.count} domain blocks... (this may take a while)")
```

Comment:
I really wish I could add a progress indicator here, but I didn't see any other progress indicators in the CLI and didn't wish to add a dependency for such

Reply:
I think there was a generic class for that added here https://github.com/mastodon/mastodon/blob/main/lib/mastodon/cli/progress_helper.rb#L18

Reply:
Okay! I've managed to use that, but did have to add in `tty-command` (same author as `tty-progress`) such that streaming the command output worked as expected from `mastodon:setup` (and I also changed the `system` calls over to using that as well)